presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/QueryBuilder.java

```diff
@@ -254,10 +256,10 @@ else if (singleValues.size() > 1) {
         return "(" + Joiner.on(" OR ").join(disjuncts) + ")";
     }
 
-    private String toPredicate(String columnName, String operator, Object value, JdbcColumnHandle column, List<TypeAndValue> accumulator)
+    private String toPredicate(String columnName, String operator, Object value, JdbcColumnHandle column, List<TypeAndValue> accumulator, Optional<Function<JdbcColumnHandle, String>> bindingPlaceholder)
```

Comment:
You can retrieve `JdbcColumnHandle ` from 4th argument `JdbcColumnHandle column`. Why don't you reuse the exising argument?

Reply:
Here it's just the function's signature. I am using the actual 4th argument  in the ```apply``` in the return clause

Reply:
Sorry, my comment was not enough. I meant we can change the type to `Optional<String> bindingPlaceholder` or boolean and then change like below. This is just my idea, thefore let's wait other member's review. 
```java
bindingPlaceholder.map(bind -> bind + column.getJdbcTypeHandle().getJdbcTypeName()).orElse("?")
```

Reply:
```Function<JdbcColumnHandle, String>``` is much generic and the connector can also pass function that results with soemthing like this: ```"cast(? as typeName)"```

Reply:
I agree with the generics. I wanted to avoid optional-functional argument. Let's follow kokosing's suggestion.

Reply:
I agree with the generics. I wanted to avoid optional-functional argument. Let's follow kokosing's suggestion.